src/nvim/regexp.c

```diff
@@ -4829,33 +4860,35 @@ static int regmatch(
         break;
 
       case RS_BRCPLX_MORE:
-        /* Pop the state.  Restore pointers when there is no match. */
+        // Pop the state.  Restore pointers when there is no match.
         if (status == RA_NOMATCH) {
           reg_restore(&rp->rs_un.regsave, &backpos);
-          --brace_count[rp->rs_no];             /* decrement match count */
+          brace_count[rp->rs_no]--;             // decrement match count
         }
         regstack_pop(&scan);
         break;
 
       case RS_BRCPLX_LONG:
-        /* Pop the state.  Restore pointers when there is no match. */
+        // Pop the state.  Restore pointers when there is no match.
         if (status == RA_NOMATCH) {
-          /* There was no match, but we did find enough matches. */
+          // There was no match, but we did find enough matches.
           reg_restore(&rp->rs_un.regsave, &backpos);
-          --brace_count[rp->rs_no];
-          /* continue with the items after "\{}" */
+          brace_count[rp->rs_no]--;
```

Comment:
increment changed here. not sure if semantics are preserved

Reply:
they are. But this file is listed in [frozen modules](https://github.com/neovim/neovim/wiki/Refactoring) because we don't plan to change it significantly. Thus we don't want to spend time on it. In particular, linting it is a low priority.